alsongDalsong/alsongDalsong/alsongDalsong/Sources/Views/SubmitAnswer/SelectAnswerView.swift

```diff
@@ -55,6 +57,11 @@ struct SelectAnswerView: View {
                             })
                             .tint(.black)
                         }
+                        .onAppear {
+                            Task {
+                                await viewModel.fetchNextSearchList(currentMusic: music)
+                            }
+                        }
```

Comment:
제가 디버깅을 해봤는데 아래와같이 매 셀마다 함수를 실행하게 되는 것 같습니다. 하지만 `Scroll offset`을 추적하는 것이 어쩌면 이것보다 더한 리소스 낭비일 수도 있겠다는 생각이 드네요. 혹시 이부분도 고려하시고 고민하시다가 선택하신 방법인가요? 궁금합니다!

<img width="200" src="https://github.com/user-attachments/assets/86c8395c-2775-484f-b237-b19f4f6f7af8" />

Reply:
우선 스유에서 방법을 찾아보다 .onAppear를 사용하면 셀이 화면에 보여짐에 따라 렌더링 되고, 그때 VM 메서드를 호출한 뒤 렌더링된 셀이 다음 페이지 기준에 맞으면 다음 데이터를 불러온다 하더라구요. 처음에 이 방법이 더 스유스럽?다 생각해서 선택했었습니다!

그리고 일단 문제라고 발견된걸 찾다 스유에서 scroll offset 추적하는 방법도 이래저래 찾아봤는데 아무래도 코드 변경이 더 많더라구요.. 그래서 조금 꺼려졌습니다 😂

Reply:
숲님 말씀처럼 셀이 화면에 보여짐에 따라서 렌더링 된다면 `리스트의 마지막 노래일때만` 가져오면 더 좋을 것 같습니다.
그러면 현준님이 말씀해주셨던 스크롤로 인한 리소스 낭비도 사용하지 않으니 더 효과적인 방법이라고 생각이듭니다..!

```suggestion
.onAppear {
      if music == viewModel.searchList.last {
           Task {
                   await viewModel.fetchNextSearchList(currentMusic: music)
            }
      }
}
```

이렇게해서 로그를 찍으면 아래처럼 나오네요!

![log](https://github.com/user-attachments/assets/ad0ee155-f854-4502-b588-89a8583d123b)